feature-libs/product-configurator/rulebased/core/state/configurator-state-utils.spec.ts

```diff
@@ -0,0 +1,257 @@
+import { ConfiguratorTestUtils } from '../../shared/testing/configurator-test-utils';
+import { Configurator } from '../model/configurator.model';
+import { ConfiguratorStateUtils } from './configurator-state-utils';
+
+const SUB_GROUP_ID = 'groupId@subGroupId';
+
+describe('ConfiguratorStateUtils', () => {
+  describe('getAttributeName', () => {
+    it('should return attribute name', () => {
+      expect(
+        ConfiguratorStateUtils['getAttributeName'](
+          SUB_GROUP_ID + '@attributeId'
+        )
+      ).toEqual('attributeId');
+    });
+  });
+
+  describe('getKey', () => {
+    it('should return a key', () => {
+      expect(
+        ConfiguratorStateUtils['getKey'](
+          'groupId@subGroupId@attributeId',
+          'attributeId'
+        )
+      ).toEqual('groupId@subGroupId');
+    });
+  });
+
+  describe('isTargeGroup', () => {
+    const groupId = 'group1';
+    const groupMatchingSupplement: Configurator.Group = {
+      id: groupId,
+      subGroups: [],
+    };
+    const groupNotMatchingSupplement: Configurator.Group = {
+      id: SUB_GROUP_ID,
+      subGroups: [],
+    };
+    const attributeSupplements: Configurator.AttributeSupplement[] = ConfiguratorTestUtils.createListOfAttributeSupplements(
+      false,
+      1,
+      0,
+      2,
+      2
+    );
+    it('should tell that a group needs to be enriched with pricing data in case group ID matches supplement key', () => {
+      expect(
+        ConfiguratorStateUtils['isTargetGroup'](
+          groupMatchingSupplement,
+          attributeSupplements
+        )
+      ).toBe(true);
+    });
+
+    it('should tell that a group does not need to be enriched if group ID does not match', () => {
+      expect(
+        ConfiguratorStateUtils['isTargetGroup'](
+          groupNotMatchingSupplement,
+          attributeSupplements
+        )
+      ).toBe(false);
+    });
+
+    it('should throw an error if supplements array is empty', () => {
+      expect(() =>
+        ConfiguratorStateUtils['isTargetGroup'](groupNotMatchingSupplement, [])
+      ).toThrowError();
+    });
+  });
+
+  describe('updateArrayElement', () => {
+    const code = 'VALUE_CODE';
+    const priceDetails: Configurator.PriceDetails = {
+      currencyIso: 'EUR',
+      value: 12,
+    };
+    let values: Configurator.Value[] = [{ valueCode: code }];
+    const valueSupplement: Configurator.ValueSupplement = {
+      attributeValueKey: code,
+      priceValue: priceDetails,
+      obsoletePriceValue: priceDetails,
+    };
+
+    it('should create new array in case predicate matches', () => {
```

Comment:
Correct me if I am wrong, but I have not seen a test that checks that the projection is done correctly and for the correct entry.
Please add this test.

Reply:
You are right. This is covered by the tests of the main method, but it I will add a test for clarity reasons

Reply:
Done